### ISSUE TITLE
When calling a procedure with an INOUT parameter of a composite type, plpgsql_check generates warnings

### ISSUE THREAD

**Issue by endijsbertans** [okbob/plpgsql_check#192]
Hello,
When calling a procedure with an INOUT parameter of a composite type, plpgsql_check generates wierd warnings. Even when I initiate the dog type with some default values it still throws the warning.

I have tried many ways to fix this warning but with no success, any ideas?
## Environment

- **Database:** PostgreSQL 16.6
- **Extension version:** 2.8.2

## Code to reproduce issue

```sql
CREATE TYPE main.dog AS (
    date_created timestamp,
    name text,
    age integer
);

-- 
CREATE OR REPLACE PROCEDURE main.initialize_dog(
    INOUT dog main.dog,
    IN dog_name text,
    IN dog_age integer
)
LANGUAGE plpgsql
AS
$body$
BEGIN
    dog.date_created := current_timestamp;
    dog.name := dog_name;
    dog.age := dog_age;
END;
$body$;

--
CREATE OR REPLACE PROCEDURE main.initialize_dog()
LANGUAGE plpgsql
AS
$body$
DECLARE
    v_dog1 main.dog;
    v_dog2 main.dog;
BEGIN
    CALL main.initialize_dog(v_dog1, 'Barky', 5);
    CALL main.initialize_dog(v_dog2, 'Barky', 5);
    CALL main.initialize_dog(v_dog1, 'Barky', 7);
END;
$body$;
```

## Screenshot

![Error output showing three warning records](https://github.com/user-attachments/assets/c7776df8-2d89-4bfa-a301-7a339c95d235)

**Comment by okbob** [comment]
should be fixed by eced9510066130c7969e44879afe7bae093ea283

**Comment by okbob** [comment]
thank you for bug report :-)

út 7. 10. 2025 v 8:04 odesílatel endijsbertans ***@***.***>
napsal:
